R/sumbetadiff.R

```diff
@@ -31,72 +40,86 @@ sumBetadiff <- function(parY, parX,
         maximum = TRUE
       )$maximum
 
-      lower <- qbetadiff(
-        p = (1 - level) / 2,
+      lower <- qbetadiff( # to recover x when F(x) is at lower percentile
+        p = (1 - coverage) / 2,
         parY = parY,
         parX = parX
       )
 
-      upper <- qbetadiff(
-        p = (1 + level) / 2,
+      upper <- qbetadiff( # to recover x when F(x) is at upper percentile
+        p = (1 + coverage) / 2,
         parY = parY,
         parX = parX
       )
-
-      ## for Go:
-      Out.go <- stats::integrate(
+      # Prob for Go:
+      auc_go <- stats::integrate(
         f = dbetadiff,
-        parY = parY,
         parX = parX,
-        lower = cutB,
+        parY = parY,
+        lower = go_cut,
         upper = 1,
         subdivisions = 1000L,
         rel.tol = .Machine$double.eps^0.1
       )$value
 
-      ## for Stop:
-      Out.nogo <- stats::integrate(
+      # Prob for Stop:
+      auc_stop <- stats::integrate(
         f = dbetadiff,
-        parY = parY,
         parX = parX,
+        parY = parY,
         lower = -1,
-        upper = cutW,
+        upper = stop_cut,
         subdivisions = 1000L,
         rel.tol = .Machine$double.eps^0.1
       )$value
 
+      assert_number(mode, upper = 1, na.ok = FALSE)
+      assert_number(lower, lower = -1, upper = 1, na.ok = FALSE)
+      assert_number(upper, lower = 0, upper = 1, na.ok = FALSE)
+      assert_number(auc_go, upper = 1, na.ok = FALSE)
+      assert_number(auc_stop, upper = 1, na.ok = FALSE)
+
       list(
         mode = mode,
         ci = c(lower, upper),
-        go = Out.go,
-        nogo = Out.nogo
+        go = auc_go,
+        stop = auc_stop
       )
     },
     silent = TRUE
   )
-
   ## if there were any errors, fall back to Monte Carlo estimation
-  if (inherits(res, "try-error")) {
+  if (inherits(result, "try-error")) { # try-error is a class
+    set.seed <- seed
```

Comment:
This would need to be
```suggestion
    set.seed(seed)
```
However, I would not like to do like this, because then calling this function has a global side effect (changing the seed) - it is better to let the user call `set.seed()` themselves before calling this function.

Reply:
ok I removed it, I wrote a note above because it may not be apparent to use a seed here